app/src/features/Preview/Preview.tsx

```diff
@@ -101,17 +106,60 @@ const Preview = (): JSX.Element => {
   const classes = useStyles();
   const { t } = useTranslation();
   const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
-  const { mappingId } = useParams<{
+  const { sourceId, mappingId } = useParams<{
+    sourceId?: string;
     mappingId?: string;
   }>();
 
+  const [previewIndex, setPreviewIndex] = useState<number>(0);
+
+  const [exploration, setExploration] = useState<
+    ExplorationResponse | undefined
+  >(undefined);
+
   const [alert, setAlert] = useState<string | undefined>(undefined);
   const handleAlertClose = () => setAlert(undefined);
 
   const [preview, setPreview] = useState<Record<string, unknown> | undefined>(
     undefined
   );
 
+  const { data: credentials } = useApiCredentialsListQuery({
+    source: sourceId,
+  });
+  const credential = credentials?.[0];
+
+  const {
+    mappings,
+    refetch: refetchMappings,
+    isMappingsFetching,
+  } = useApiSourcesExportRetrieveQuery(
+    { id: sourceId ?? "" },
+    {
+      skip: !sourceId,
+      selectFromResult: ({ data: mappings, isFetching }) => {
+        const resource = mappings?.resources?.find(
+          ({ id }) => id === mappingId
+        );
+        return {
+          mappings: mappings &&
+            credential && {
+              ...mappings,
+              credential: {
+                ...mappings.credential,
+                login: credential.login,
+                password: credential.password,
+              },
+              resources: resource && [resource],
+            },
+          isMappingsFetching: isFetching,
```

Comment:
Does returning `isFetching` as `isMappingsFetching` necessary ? If we don't do this operation, we can still get `isFetching` as usual ? (line 135)

Reply:
nop we can't :/